src/app/body/product/tabs/tabs.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, ContentChildren, QueryList, AfterContentInit } from '@angular/core';
+import { TabComponent } from '../tab/tab.component';
+
+@Component({
+  selector: 'app-tabs',
+  templateUrl: './tabs.component.html',
+  styleUrls: ['./tabs.component.scss']
+})
+export class TabsComponent implements AfterContentInit {
+  @ContentChildren(TabComponent) tabs: QueryList<TabComponent>;
+  constructor() { }
+
+  ngAfterContentInit(): void {
+    const activeTabs = this.tabs.filter((tab) => tab.active);
+    if (activeTabs.length === 0) {
+      this.selectTab(this.tabs.first);
+    }
+  }
+
+  selectTab(tab): void {
```

Comment:
could you please add 'public' or 'private' for each method

Reply:
fixed